fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -521,13 +524,61 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
      * thread, early in the app startup sequence.
      */
     private fun beginSetupMegazord() {
-        // Note: Megazord.init() must be called as soon as possible ...
-        Megazord.init()
+        try {
+            // Note: Megazord.init() must be called as soon as possible ...
+            Megazord.init()
+
+            initializeRustErrors(components.analytics.crashReporter)
+            // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
 
-        initializeRustErrors(components.analytics.crashReporter)
-        // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
+            RustLog.enable()
+        } catch (e: UnsatisfiedLinkError) {
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                reportUnsatisfiedLinkErrorBreadcrumbs()
+            } catch (e: Throwable) {
+                // This shouldn't happen, but if it does it's better to ignore the exception from
+                // the breadcrumb code and rethrow the initial exception.
+            }
+            throw e
+        }
+    }
 
-        RustLog.enable()
+    private fun reportUnsatisfiedLinkErrorBreadcrumbs() {
+        val breadcrumbs = mutableListOf<String>()
+        val apkPath = applicationContext.getApplicationInfo().sourceDir
+        breadcrumbs.add("APK: $apkPath")
+        val apkDir = File(apkPath).getParentFile()
+        val installSourcePackage = if (SDK_INT >= Build.VERSION_CODES.R) {
+            packageManager.getInstallSourceInfo(packageName).installingPackageName
+        } else {
+            @Suppress("DEPRECATION")
+            packageManager.getInstallerPackageName(packageName)
+        }
+        breadcrumbs.add("Installer package name: $installSourcePackage")
+
+        val apkDirFileList = apkDir?.let {
+            it.walk()
+                .filter { it != apkDir }
+                .map { it.relativeTo(apkDir).toString() }
+                .joinToString(", ")
+        }
+        breadcrumbs.add("Files inside dir: $apkDirFileList")
+
+        val apkFileList = ZipInputStream(FileInputStream(apkPath)).use {
```

Comment:
Does this have privacy issue?

Reply:
I'm not sure how to judge that, but it feels like technical data to me and I think that can be shared on Sentry.  IMO, the most sensitive would be the installer package name, maybe some users wouldn't want us to know how they installed the app?  I personally don't think that's an issue, but I'm no expert.  Who is a good person to ask about this.

Reply:
The installer package name is also captured in the [startup metrics](https://github.com/mozilla-mobile/firefox-android/blob/da34eb224eb391b21dd2e7ff58bdf082a98fffc4/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L847-L854), maybe this means it's okay to share in a sentry report.

Reply:
Can this crash only happen on fresh install? or can it happen at any time when user starts the app?